sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Search.cs

```diff
@@ -378,8 +400,8 @@ internal async Task<List<Document>> GetNextSetHelperAsync(CancellationToken canc
 
                         Dictionary<string, Condition> keyConditions, filterConditions;
                         SplitQueryFilter(Filter, SourceTable, queryReq.IndexName, out keyConditions, out filterConditions);
-                        queryReq.KeyConditions = keyConditions;
-                        queryReq.QueryFilter = filterConditions;
+                        queryReq.KeyConditions = keyConditions.Count > 0 ? keyConditions : null;
```

Comment:
Can `keyConditions` or `filterConditions` be null here? Earlier in this file you used `keyConditions?.Count` for example.